source/simulator/stokes_matrix_free.cc

```diff
@@ -1460,6 +1466,10 @@ namespace aspect
 
     LinearAlgebra::BlockVector stokes_rhs_correction (sim.introspection.index_sets.stokes_partitioning, sim.mpi_communicator);
     internal::ChangeVectorTypes::copy(stokes_rhs_correction,rhs_correction);
+
+    if (sim.do_pressure_rhs_compatibility_modification)
+      sim.make_pressure_rhs_compatible(stokes_rhs_correction);
```

Comment:
I do not quite understand this line. Shouldnt you first add the correction to the system_rhs and then make the new system_rhs compatible? Or does the order of operations not matter, because if system_rhs is already compatible, and you add a compatible correction the results is still compatible? I do not exactly remember how the correction looks like.

Reply:
> Or does the order of operations not matter, because if system_rhs is already compatible, and you add a compatible correction the results is still compatible?

I think that is correct. @tcclevenger please explain and put it into a comment above this line of code.

Reply:
I've changed this up so that we find the rhs correction and add it to the `system_rhs` in  `StokesMatrixFree`, and then do the compatibility function call in assembly.cc _only after_ calling `correct_stokes_rhs()`. This removes the need to do this 2x.